vizro-core/src/vizro/models/_components/container.py

```diff
@@ -34,9 +37,31 @@ class Container(VizroBaseModel):
     )
     title: str = Field(description="Title to be displayed.")
     layout: Annotated[Optional[Layout], AfterValidator(set_layout), Field(default=None, validate_default=True)]
+    theme: Optional[Literal["filled", "outlined"]] = Field(
+        default=None,
+        description="Predefined styles to choose from. Options are `filled` or `outlined`. Defaults to `None`.",
+    )
```

Comment:
I'm not absolutely 100% sure about this but like 90% sure that it would make more sense if we give the container default theme a name even if it's just "none". 

e.g. if we think of applying these styles to `vm.Card`, let's say we have the same 3 styles available but the default is "filled". Then what would options look like for `vm.Card` if they wanted to be able to select the same styling as is the default for a container: `theme=filled/outlined/???". We'd need a name for "???". And it's maybe a bit confusing if `theme=None` means something different for `vm.Card` vs. `vm.Container`. It would make more sense for the options to be the same but the default to be different.

```suggestion
    theme: Optional[Literal["none", "filled", "outlined"]] = Field(
        default="none",
        description="Predefined styles to choose from. Options are `none`, `filled` or `outlined`. Defaults to `none`.",
    )
```

Or do we even have a good description of what the default is other than `"none"`? Like "transparent" or something?

Reply:
Copied text from https://github.com/mckinsey/vizro/pull/1002#pullrequestreview-2653509111 to continue discussion here in this thread

---

First of all, I'm so happy we abandoned the `background=True/False` idea! 😅 

As `theme` feels misleading, I'll use `style_variant` in the rest of the comment as it feels  much more natural to me 😄 

Before merging this PR, I think we should establish a rough plan for `style_variant`. In future I see that we will upgrade this solution to another level and that we will have a predefined set like `"transparent"`, `"filled"`, `"outlined"`, `"elevated"`, `"rounded"`... In that case each component could have a default like for example, `"transparent"` for `Container`, `"filled"` for `Card` and `Alert`, while `Button` might include additional options like `"link"` and `"disabled"`.

Open questions:
1. Should we default Container to `"transparent"` instead of `None`?
2. Should we align our `style_variant` values with existing standards like Bootstrap, or define our own? Caution: users could require something like `"rounded-elevated"` 🤔 

While we don’t need a full plan before merging, having a clear direction will help ensure consistency. Looking forward to your thoughts! 🚀

Reply:
FYI @petar-qb there's plans for "elevated" but that's about it. We don't intend to enable lots of different styles like rounded etc. here since they're outside our designs. I think there should always be less than 5 or so options. In future users might be able to fully customise their theme at a global level (e.g. if they want to set `rounded=True` throughout their app). It's already possible at a component level to do arbitrary customisation using CSS or the new `extra` but we don't intend to make it easy to get any style you want through this `variant` argument. It's deliberately quite restricted to things that we think will look good.

As for aligning with bootstrap, we don't want to fully match bootstrap. Users shouldn't need to read bootstrap docs to understand how to change this, and descriptive terms like `filled` are more declarative and implementation-independent. Think of our `variant` argument as a kind of higher-level translation layer into bootstrap (which is already a shortcut for CSS). And advanced users can access the full range of bootstrap configuration still through `extra` if they want to do something that's outside our very limited translation layer.

Reply:
Just saw this discussion and thought I'd chip in. I am actually against `None` meaning different things, I think it is confusing. In general working with the schema recently I think that `None` causes often more trouble than not.

If we are able to give a name to every default of components that have a `appearance/style` argument, then I think we should not rely on None. I think it is very ok to have the card be `filled` by default, and the container to be `transparent`. To me that is clear and concise.

So in summary:
- no `None`
- definitely no `"none"`
- ideally a default that is descriptive, like `transparent` or `filled` or whatever for the respective component

Reply:
@maxschulz-COL generally agree with you that we should avoid `None` (this was my original point in this thread). Curious why do you think `"none"` is even worse? It's just a more confusing version of `None`?

Overall @petar-qb, @maxschulz-COL and I are all agreed that if we can find a name that accurately describes the style, like "transparent" or "plain" then that is best though. @huong-li-nguyen wdyt? The only argument I have against "transparent" is that "outlined" is also transparent. But maybe "plain" just sounds silly.

Reply:
Yes, I think it is almost a weaker version of `"none"` and somewhat reminiscent of the `ALL` issue. Because it is also not really true that there is no styling (see `Card` or even `Container` (still has the dashboard background and potentially padding)) - so in order to avoid potential misunderstandings I think a dedicated description is cleanest.

At first I thought `default` might be good, but it somewhat fails for the same reasons...

Reply:
I'm going with plain. I don't think your idea is silly at all, @antonymilne. It's also what MUI does on a global level for variants: https://mui.com/joy-ui/main-features/global-variants/

- solid (default), soft, outlined, and plain.

Just to mention some other alternatives I've seen in UI frameworks: plain, text, standard 😄 

I'm not a fan of `default` because the actual default varies per component. For instance, our default for the Card is "elevated," for the Container it's "plain," and for the Button it's "primary." Using "default" would be quite confusing as it would not be the default for every component.

Reply:
I always understood it more like `default` means just what we decide, so basically `vizro`. But yes, I am also against it because it clearly is confusing!